### PR TITLE
osm_editor_hits_fitching_update

### PR DESCRIPTION
**Pull request by nnngrach**
[Issue](https://github.com/osmandapp/OsmAnd/issues/13660)

Screenshot 2022-12-05 at 18 33 15 Screenshot 2022-12-05 at 18 33 25

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/AdvancedEditPoiFragment.java [hunk 1]
```diff
@@ -374,4 +381,51 @@ public static void addPoiToStringSet(AbstractPoiType abstractPoiType, Set<String
...
+
+	private class OsmTagsFilter extends Filter {
+
+		@Override
+		protected FilterResults performFiltering(CharSequence constraint) {
+			FilterResults results = new FilterResults();
+			List<String> filteredTags = new ArrayList<String>();
+			String query = constraint.toString().trim();
+			for (String tag : allTags) {
+				if (tag.startsWith(query) || tag.startsWith("service:" + query)) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/AdvancedEditPoiFragment.java:410]: "service:" ??? and what if it is other tag

too complicated

**Comment by vshcherb** [comment]
Still quite a lot of code, I'm not sure it's optimal solution.

File: OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/AdvancedEditPoiFragment.java [hunk 2]
```diff
@@ -374,4 +381,51 @@ public static void addPoiToStringSet(AbstractPoiType abstractPoiType, Set<String
...
+			results.values = filteredTags;
+			results.count = filteredTags.size();
+			return results;
+		}
+
+		@SuppressWarnings("unchecked")
+		@Override
+		protected void publishResults(CharSequence constraint, FilterResults results) {
+			if (results.values != null) {
+				synchronized (mAdapter) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/AdvancedEditPoiFragment.java:423]: publishResults runs in UI Thread probably it's not neded

Comment (alex-osm) [reply OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/AdvancedEditPoiFragment.java:423]: Moreover `mAdapter` is not filter's adapter

**Pull request merged**